R/utils.R

```diff
@@ -122,3 +122,39 @@ shouldEval <- function(options) {
 }
 # JSEvals(list(list(foo.bar=JS("hi"), baz.qux="bye"))) == "0.foo\\.bar"
 
+#' Render widget as an iframe
+#'
+#' This function allows a widget to be embedded in an Rmd document as an iframe.
+#' This is especially useful when there is potential for conflicts between
+#' assets on the parent page and those used by the widget.
+#'
+#' @inheritParams saveWidget
+#' @param width width of iframe
+#' @param height height of iframe
+#' @param border-width width of the borders. default is 0px
+#' @param ... css styles to add to the iframe
+#' @export
+as.iframe <- function(widget, file = NULL, selfcontained = TRUE,
+    libdir = NULL, width = '100%', height = 500, `border-width` = '0px', ...){
+  if (is.null(file)){
+    file <- basename(tempfile(fileext = '.html'))
```

Comment:
1. The filename matters only for the option `selfcontained = FALSE` in which case it makes more sense to save it relative to the directory of the `rmd` file. 
2. Even without `libdir`, I think `selfcontained = F` makes sense, since it will save the widget html to  file and link dependencies from the default value of `libdir`, which I think is the same as the widget name.
3. I don't think we want to unlink the file, especially when `selfcontained = FALSE`, since it is referred to by the `iframe`. We can unlink it for `selfcontained = TRUE`.